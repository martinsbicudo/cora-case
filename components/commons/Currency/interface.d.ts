export interface CurrencyProps {
  type?: 'credit' | 'debit' | 'refunded'
  value: number
}

export interface StyledCurrencyProps {
  type: 'credit' | 'debit' | 'refunded'
}
