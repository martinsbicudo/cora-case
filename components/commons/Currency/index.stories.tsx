import { StoryObj, Meta } from '@storybook/react'

import Currency from '.'

export default {
  title: 'Commons / Currency',
  component: Currency,
  argTypes: {
    type: {
      control: 'select',
      defaultValue: 'credit',
      options: ['credit', 'debit', 'refunded'],
    },
    value: {
      description: 'Value in cents',
    },
  },
} as Meta

export const Default: StoryObj = {
  args: {
    value: 8000,
  },
}
