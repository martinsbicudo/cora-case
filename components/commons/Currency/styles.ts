import styled, { css } from 'styled-components'

import { StyledCurrencyProps } from './interface'

export const Currency = styled.span<StyledCurrencyProps>`
  font-size: 1.6rem;
  line-height: 2.4rem;
  font-weight: 900;

  ${({ theme, type }) => {
    switch (type) {
      case 'debit':
        return css`
          color: ${theme.colors.primary};
        `
        break
      case 'credit':
        return css`
          color: ${theme.colors.secondary};
        `
        break
      default:
        return css`
          color: ${theme.colors['off-black']};
          text-decoration: line-through;
        `
        break
    }
  }};
`
