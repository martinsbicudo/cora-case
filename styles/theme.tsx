const theme = {
  colors: {
    primary: '#FE3E6D',
    secondary: '#1A93DA',
    'soft-secondary': '#EAF7FF',
    'off-black': '#3B3B3B',
    'gray-1': '#6B7076',
    'gray-2': '#C7CBCF',
    'gray-3': '#DEE4E9',
    'gray-4': '#F0F4F8',
    white: '#FFFFFF',
  },
}

export default theme
