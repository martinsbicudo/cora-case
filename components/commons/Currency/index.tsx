import { currencyFormatter } from '@/lib'

import { CurrencyProps } from './interface'
import * as Styled from './styles'

const InputFilter = ({ type = 'credit', value }: CurrencyProps) => {
  const getCurrency = () => {
    const currency = currencyFormatter(value)
    return type === 'debit' ? `- ${currency}` : currency
  }

  return <Styled.Currency type={type}>{getCurrency()}</Styled.Currency>
}

export default InputFilter
