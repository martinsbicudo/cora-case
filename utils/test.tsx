import type { ReactElement, ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'

import theme from '@/styles/theme'
import { render, RenderOptions } from '@testing-library/react'

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export { customRender as render }
