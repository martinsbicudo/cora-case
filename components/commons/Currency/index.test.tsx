import theme from '@/styles/theme'
import { render } from '@/test/helpers'

import Currency from '.'
import { CurrencyProps } from './interface'

describe('Components / Currency', () => {
  const renderComponent = (type?: CurrencyProps['type']) =>
    render(<Currency type={type} value={8000} />)

  it('should render component', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should find credit type style', () => {
    const { container } = renderComponent()
    expect(container.firstChild).toHaveStyle(`color: ${theme.colors.secondary}`)
  })

  it('should find debit type style', () => {
    const { container } = renderComponent('debit')
    expect(container.firstChild).toHaveStyle(`color: ${theme.colors.primary}`)
  })

  it('should find refunded type style', () => {
    const { container } = renderComponent('refunded')
    expect(container.firstChild).toHaveStyle(
      `color: ${theme.colors['off-black']}`
    )
  })
})
