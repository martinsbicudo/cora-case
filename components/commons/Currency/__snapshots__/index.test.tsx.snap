// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Currency should render component 1`] = `
<div>
  <p
    class="sc-bcXHqe hrDNvr"
    type="credit"
  >
    R$ 80,00
  </p>
</div>
`;
